core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -261,4 +267,55 @@ private Connector createConnector(ConnectionConfig config) {
         refreshTimeoutMs,
         serverProxyPort);
   }
+
+  private ConnectionConfig resolveConfig(ConnectionConfig config) {
+    if (config.getNamedConnection() != null) {
+      ConnectorKey namedConfig = resolveKey(config);
+      if (namedConfig == null) {
+        throw new IllegalArgumentException(
+            "Named configuration " + config.getNamedConnection() + " does not exist.");
+      }
+      return namedConfig.getConfig();
+    }
+    return config;
+  }
+
+  private ConnectorKey resolveKey(ConnectionConfig config) {
+    Optional<ConnectorKey> namedConfig =
```

Comment:
rename to `connectorKey` or something similar

Reply:
Fixed.